### PR TITLE
Bump recommended Git version, fix prune-tags for older versions

### PR DESCRIPTION
**Pull request by kuleshovilya**
Bump recommended Git version notification to 2.17.*
Fix --prune-tags for versions older than 2.17, since it's not supported
Related [issue](https://github.com/microsoft/azure-pipelines-agent/issues/3572)

### CODE REVIEW

File: src/Agent.Worker/Build/GitCommandManager.cs [hunk 1]
```diff
@@ -232,8 +232,8 @@ public async Task<int> GitFetch(IExecutionContext context, string repositoryPath
                 refSpec = refSpec.Where(r => !string.IsNullOrEmpty(r)).ToList();
             }
 
-            // insert prune-tags if knob is false to sync tags with the remote
-            string pruneTags = AgentKnobs.DisableFetchPruneTags.GetValue(context).AsBoolean() ? string.Empty : "--prune-tags";
+            // insert prune-tags if knob is false to sync tags with the remote and Git version is above 2.17
```
Comment (anatolybolshakov) [inline src/Agent.Worker/Build/GitCommandManager.cs:235]:
```suggestion
            // insert prune-tags if DisableFetchPruneTags knob is false and Git version is above 2.17
```

**Review by anatolybolshakov — APPROVED** [review]

File: src/Agent.Worker/Build/GitCommandManager.cs (hunk 1 above)
Comment (kuleshovilya) [reply src/Agent.Worker/Build/GitCommandManager.cs:235]: Yes, this wording does seems better

File: src/Agent.Plugins/GitCliManager.cs [hunk 2]
```diff
@@ -175,7 +175,7 @@ public async Task<int> GitFetch(AgentTaskPluginExecutionContext context, string
             string progress = reducedOutput ? string.Empty : "--progress";
 
             // insert prune-tags if knob is false to sync tags with the remote
-            string pruneTags = AgentKnobs.DisableFetchPruneTags.GetValue(context).AsBoolean() ? string.Empty : "--prune-tags";
+            string pruneTags = !EnsureGitVersion(new Version(2, 17), throwOnNotMatch: false) ? string.Empty : AgentKnobs.DisableFetchPruneTags.GetValue(context).AsBoolean() ? string.Empty : "--prune-tags";
```
Comment (DenisRumyantsev) [inline src/Agent.Plugins/GitCliManager.cs:178]: Can we make it a little more readable?

```suggestion
            string pruneTags = string.Empty;
            if (EnsureGitVersion(new Version(2, 17), throwOnNotMatch: false) && !AgentKnobs.DisableFetchPruneTags.GetValue(context).AsBoolean())
            {
                pruneTags = "--prune-tags";
            }
```

File: src/Agent.Worker/Build/GitCommandManager.cs [hunk 3]
```diff
@@ -232,8 +232,8 @@ public async Task<int> GitFetch(IExecutionContext context, string repositoryPath
                 refSpec = refSpec.Where(r => !string.IsNullOrEmpty(r)).ToList();
             }
 
-            // insert prune-tags if knob is false to sync tags with the remote
-            string pruneTags = AgentKnobs.DisableFetchPruneTags.GetValue(context).AsBoolean() ? string.Empty : "--prune-tags";
+            // insert prune-tags if DisableFetchPruneTags knob is false and Git version is above 2.17
+            string pruneTags = !EnsureGitVersion(new Version(2, 17), throwOnNotMatch: false) ? string.Empty : AgentKnobs.DisableFetchPruneTags.GetValue(context).AsBoolean() ? string.Empty : "--prune-tags";
```
Comment (DenisRumyantsev) [inline src/Agent.Worker/Build/GitCommandManager.cs:236]: Here is exactly the same:

```suggestion
            string pruneTags = string.Empty;
            if (EnsureGitVersion(new Version(2, 17), throwOnNotMatch: false) && !AgentKnobs.DisableFetchPruneTags.GetValue(context).AsBoolean())
            {
                pruneTags = "--prune-tags";
            }
```

**Comment by DenisRumyantsev** [comment]
`src/Agent.Plugins/GitCliManager.cs` and `src/Agent.Worker/Build/GitCommandManager.cs` have a lot of duplicate code, specifically: `EnsureGitVersion`, `EnsureGitLFSVersion`, `LoadGitExecutionInfo`, `GitInit` and other methods. It would be better to move out them to the separate module (perhaps not in this PR).

File: src/Agent.Worker/Build/GitCommandManager.cs (hunk 3 above)
Comment (kuleshovilya) [reply src/Agent.Worker/Build/GitCommandManager.cs:236]: Yes, seems good

**Review by DenisRumyantsev — APPROVED** [review]
LGTM, thanks!

**Review by anatolybolshakov — APPROVED** [review]

**Comment by Mertsch** [comment]
Shouldn't it be `Deprecated` instead of `Depricated`

**Pull request merged**